src/protocol/resp/zrange.rs

```diff
@@ -0,0 +1,157 @@
+    response_buf: &mut Vec<u8>,
+    req: &SortedSetRange,
+) -> ProxyResult {
+    update_method_metrics(&ZRANGE, &ZRANGE_EX, async move {
+        if *req.range_type() == RangeType::ByLex {
+            klog_1(&"zrange", &req.key(), Status::ServerError, 0);
+            return Err(ProxyError::from(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "Momento proxy does not support BYLEX for ZRANGE",
+            )));
+        }
+
+        let response = match *req.range_type() {
+            RangeType::ByIndex => {
+                // ByIndex accepts only integers as inclusive start and inclusive stop values
+                let start = parse_score_boundary_as_integer(req.start())?;
+                let stop = parse_score_boundary_as_integer(req.stop())?;
+
+                let order = match req.optional_args().reversed {
+                    Some(true) => SortedSetOrder::Descending,
+                    _ => SortedSetOrder::Ascending,
+                };
+
+                match time::timeout(
+                    Duration::from_millis(200),
+                    client.sorted_set_fetch_by_rank(
+                        cache_name,
+                        req.key(),
+                        order,
+                        Some(start),
+                        Some(stop),
+                    ),
+                )
+                .await
+                {
+                    Ok(Ok(r)) => r,
+                    Ok(Err(e)) => {
+                        klog_1(&"zrange", &req.key(), Status::ServerError, 0);
+                        return Err(ProxyError::from(e));
+                    }
+                    Err(e) => {
+                        klog_1(&"zrange", &req.key(), Status::Timeout, 0);
+                        return Err(ProxyError::from(e));
+                    }
+                }
+            }
+            RangeType::ByScore => {
+                // ByScore can accept start/stop values: `(` for exclusive boundary, `+inf`, `-inf`.
+                let (start, exclusive_start) = parse_score_boundary_as_float(req.start())?;
+                let (stop, exclusive_stop) = parse_score_boundary_as_float(req.stop())?;
+
+                let order = match req.optional_args().reversed {
+                    Some(true) => SortedSetOrder::Descending,
+                    _ => SortedSetOrder::Ascending,
+                };
+
+                // Momento accepts only inclusive min and max scores, so we add one if the boundary is exclusive,
+                // so we add a small epsilon if the boundary is exclusive.
+                const EPSILON: f64 = 1e-9; // Small value to adjust for exclusive boundaries
+                let fetch_request = SortedSetFetchByScoreRequest::new(cache_name, req.key())
+                    .order(order)
+                    .min_score(start + if exclusive_start { EPSILON } else { 0.0 })
+                    .max_score(stop - if exclusive_stop { EPSILON } else { 0.0 })
```

Comment:
Same situation here: the service supports declaring the bounds inclusive vs exclusive. We just need to add that to the Rust SDK.